paimon-core/src/main/java/org/apache/paimon/table/source/DataFilePlan.java

```diff
@@ -20,12 +20,13 @@
 
 import org.apache.paimon.table.source.snapshot.StartingScanner;
 
+import java.io.Serializable;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 
 /** Scanning plan containing snapshot ID and input splits. */
-public class DataFilePlan implements TableScan.Plan {
+public class DataFilePlan implements TableScan.Plan, Serializable {
```

Comment:
Why here need to be `Serializable`?

Reply:
Because we need to serialize `FilesSplit`, and we don't put `FileStoreTable `in `FilesSplit`, then we need to put plan in it. `TableScan.Plan` will be serialized when `FilesSplit` serializing.  Same for `PartitionsSplit`.

Reply:
No, you just need `List<Split>`.